### PR TITLE
[Doc] Add 3.2 http sql

### PR DESCRIPTION
**Pull request by evelynzhaojie**
Why I'm doing:

What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/reference/HTTP_API/SQL.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+#### Succeeded
+
+Each row in the response message is a JSON object. JSON objects separated by `\n`.
+
+- For a SELECT statement, the following JSON objects are returned.
+
+| Object       | Description                                                  |
+| ------------ | :----------------------------------------------------------- |
+| `connectionId` | Connection ID. You can cancel a query this is pending for a long time by calling KILL `<connectionId>`. |
```
Comment (EsoragotoSpirit) [inline docs/en/reference/HTTP_API/SQL.md:116]: this is --> if it is

File: docs/en/reference/HTTP_API/SQL.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+## Description
+
+StarRocks v3.2.0 introduces the HTTP SQL API for users to perform various types of queries using HTTP. Currently, this API supports SELECT, SHOW, EXPLAIN, and KILL statements.
+
+Syntax using the curl command:
+
+```shell
+curl -X POST 'http://<fe_ip>:<fe_http_port>/api/v1/catalogs/<catalog_name>/databases/<database_name>/sql' \
+   -u '<username>:<password>'  -d '{"query": "<sql_query>;", "sessionVariables":{"var_name":<var_value>}}' \
```
Comment (EsoragotoSpirit) [inline docs/en/reference/HTTP_API/SQL.md:15]: `"<var_name>"`?

File: docs/en/reference/HTTP_API/SQL.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+```sql
+POST 'http://<fe_ip>:<fe_http_port>/api/v1/catalogs/<catalog_name>/databases/<database_name>/sql'
+```
+
+| Field                    | Description                                                  |
+| ------------------------ | :----------------------------------------------------------- |
+|  fe_ip                   | FE node IP address.                                                  |
+|  fe_http_port            | FE HTTP port.                                           |
+|  catalog_name            | The catalog name. Currently, this API supports querying only internal tables, which means `<catalog_name>` can only be set to `default_catalog`. |
+|  database_name           | The database name. If no database name is specified and a table is used in the SQL query, you must prefix the table name with its database name, for example, `database_name.table_name`. |
```
Comment (EsoragotoSpirit) [inline docs/en/reference/HTTP_API/SQL.md:32]: If ~~no database name is specified and~~ a table is used in the SQL query, you must prefix the table name with its database name, for example, `database_name.table_name`.

File: docs/en/reference/HTTP_API/SQL.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+```SQL
+Authorization: Basic <credentials>
+```
+
+Basic authentication is used, that is, enter the username and password for `credentials` (`-u '<username>:<password>'`). If no password is set for the username, you can pass in only the username and leave the password empty. For the root account with no password, you can enter `-u 'root:'`.
+
+### Request body
+
+```SQL
+-d '{"query": "<sql_query>;", "sessionVariables":{"var_name":<var_value>}}'
```
Comment (EsoragotoSpirit) [inline docs/en/reference/HTTP_API/SQL.md:57]: `"<var_name>"`?

File: docs/en/reference/HTTP_API/SQL.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+curl -X POST 'http://<fe_ip>:<fe_http_port>/api/v1/catalogs/<catalog_name>/databases/<database_name>/sql' \
+   -u '<username>:<password>'  -d '{"query": "<sql_query>;", "sessionVariables":{"var_name":<var_value>}}' \
+   --header "Content-Type: application/json"
+```
+
+## Request message
+
+### Request line
+
+```sql
```
Comment (EsoragotoSpirit) [inline docs/en/reference/HTTP_API/SQL.md:23]: I think all the code blocks are shell except the response body examples

File: docs/en/reference/HTTP_API/SQL.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+{"data":[2,"30.00","35.00"]}
+{"statistics":{"scanRows":0,"scanBytes":0,"returnRows":5}}
+```
+
+### Cancel a query
+
+To cancel a query that run an unexpected long time, you can close the connection. StarRocks will cancel this query when it detects the connection is closed.
+
+You can also call KILL `connectionId` to cancel this query:
+
```
Comment (EsoragotoSpirit) [inline docs/en/reference/HTTP_API/SQL.md:174]: For example:

File: docs/en/reference/HTTP_API/SQL.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+   POST /api/v1/catalogs/<catalog_name>/databases/<database_name>/sql
+   ```
+
+### Authentication method
+
+```SQL
+Authorization: Basic <credentials>
+```
+
+Basic authentication is used, that is, enter the username and password for `credentials` (`-u '<username>:<password>'`). If no password is set for the username, you can pass in only the username and leave the password empty. For the root account with no password, you can enter `-u 'root:'`.
```
Comment (EsoragotoSpirit) [inline docs/en/reference/HTTP_API/SQL.md:52]: you can pass in only the username and a colon, and leave ...

File: docs/zh/reference/HTTP_API/SQL.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+| ------------ | :----------------------------------------------------------- |
+| `connectionId` | 该连接对应的 ID，可通过 KILL `<connectionId>` 来取消某个执行时间过长的 query。 |
+| `meta`        | 用来描述每个列。Key 为 meta, Value 为一个 JSON array。array 中的每个 object 表示一个 column。 |
+| `data`         | 代表一行数据。Key 为 data, Value 为一个 JSON  array，array 中包含一条数据。 |
+| `statistics`   | 本次执行结果的统计信息。                                       |
+
+- 对于 SHOW，会返回 `meta`，`data`，`statistics`。
+- 对于 EXPLAIN，会返回一个 `explain` 对象，展示这条查询详细的执行计划。
+
+样例：这里换行以 `\n` 表示。发送时 StarRocks 使用 HTTP chunked 方式传输数据，FE 每获取一批数据，就将该批数据流式转发给客户端。客户端可以按行解析 StarRocks 发送的数据，而不需要缓存现有结果直到所有数据发送完毕再进行解析。这可以降低客户端的内存消耗。
```
Comment (EsoragotoSpirit) [inline docs/zh/reference/HTTP_API/SQL.md:124]: 此处需要放在英文吗

**Review by EsoragotoSpirit — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Pull request merged**